integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/action/credential/CredentialTestAction.java

```diff
@@ -0,0 +1,89 @@
+package com.sequenceiq.it.cloudbreak.newway.action.credential;
+
+import static com.sequenceiq.it.cloudbreak.newway.log.Log.log;
+import static com.sequenceiq.it.cloudbreak.newway.log.Log.logJSON;
+import static java.lang.String.format;
+
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.credentials.responses.CredentialV4Response;
+import com.sequenceiq.it.cloudbreak.newway.CloudbreakClient;
+import com.sequenceiq.it.cloudbreak.newway.context.TestContext;
+import com.sequenceiq.it.cloudbreak.newway.entity.credential.CredentialTestDto;
+
+public class CredentialTestAction {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CredentialTestAction.class);
+
+    private CredentialTestAction() {
+
+    }
+
+    public static CredentialTestDto createIfNotExist(TestContext testContext, CredentialTestDto entity, CloudbreakClient client) throws Exception {
+        LOGGER.info("Create Credential with name: {}", entity.getRequest().getName());
+        try {
+            entity.setResponse(
+                    client.getCloudbreakClient().credentialV4Endpoint().post(client.getWorkspaceId(), entity.getRequest())
+            );
+            logJSON(LOGGER, "Credential created successfully: ", entity.getRequest());
+        } catch (Exception e) {
```

Comment:
catching `ProxyMethodInvocationException` is sufficient since any endpoint call should go through the `BeforeAfterMessagingProxyExecutor` which catches and converts all `Exception` to a `ProxyMethodInvocationException`